internal/scm/bitbucketcloud/bitbucket_cloud.go

```diff
@@ -0,0 +1,324 @@
+package bitbucketcloud
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"slices"
+	"strings"
+
+	"github.com/ktrysmt/go-bitbucket"
+	"github.com/lindell/multi-gitter/internal/scm"
+	"github.com/pkg/errors"
+)
+
+type BitbucketCloud struct {
+	repositories []string
+	workspaces   []string
+	users        []string
+	username     string
+	token        string
+	sshAuth      bool
+	httpClient   *http.Client
+	bbClient     *bitbucket.Client
+}
+
+func New(username string, token string, repositories []string, workspaces []string, users []string, sshAuth bool, transportMiddleware func(http.RoundTripper) http.RoundTripper) (*BitbucketCloud, error) {
+	if strings.TrimSpace(token) == "" {
+		return nil, errors.New("bearer token is empty")
+	}
+
+	bitbucketCloud := &BitbucketCloud{}
+	bitbucketCloud.repositories = repositories
+	bitbucketCloud.workspaces = workspaces
+	bitbucketCloud.users = users
+	bitbucketCloud.username = username
+	bitbucketCloud.token = token
+	bitbucketCloud.sshAuth = sshAuth
+	bitbucketCloud.httpClient = &http.Client{
+		Transport: transportMiddleware(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: false}, // nolint: gosec
+		}),
+	}
+	bitbucketCloud.bbClient = bitbucket.NewBasicAuth(username, token)
+
+	return bitbucketCloud, nil
+}
+
+func (bbc *BitbucketCloud) CreatePullRequest(ctx context.Context, repo scm.Repository, prRepo scm.Repository, newPR scm.NewPullRequest) (scm.PullRequest, error) {
+
+	splitRepoFullName := strings.Split(prRepo.FullName(), "/")
+
+	repoOptions := &bitbucket.RepositoryOptions{
+		Owner:    bbc.workspaces[0],
+		RepoSlug: splitRepoFullName[1],
+	}
+	defaultReviewers, _ := bbc.bbClient.Repositories.Repository.ListDefaultReviewers(repoOptions)
+	for _, reviewer := range defaultReviewers.DefaultReviewers {
+		newPR.Reviewers = append(newPR.Reviewers, reviewer.Uuid)
+	}
+
+	prOptions := &bitbucket.PullRequestsOptions{
```

Comment:
Can pull request happen from a fork in bbc? In that case I don't think the owner here is correct?

Reply:
We need to add a parameter to allow users to specify a bitbucket "project" to fork into before we can specify the owner fully here. 

We were hoping to do that as a separate PR and document that as a known limitation but let us know if that is a blocker we should resolve it before we get this change merged.

Reply:
I'm fine with merging something that does not have all functionality of all other systems 🙂 But please list these, as I will otherwise make comments on things that will not be relevant to you

Reply:
I made a note in the README template about bitbucket cloud support and the current limitations that should help clarify things.